core/src/main/resources/org/broadinstitute/dsde/rawls/liquibase/changesets/20210615_sharded_entity_tables.xml

```diff
@@ -0,0 +1,271 @@
+<?xml version="1.0" encoding="UTF-8" standalone="no"?>
+<databaseChangeLog logicalFilePath="dummy" xmlns="http://www.liquibase.org/xml/ns/dbchangelog" xmlns:ext="http://www.liquibase.org/xml/ns/dbchangelog-ext" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://www.liquibase.org/xml/ns/dbchangelog-ext http://www.liquibase.org/xml/ns/dbchangelog/dbchangelog-ext.xsd http://www.liquibase.org/xml/ns/dbchangelog http://www.liquibase.org/xml/ns/dbchangelog/dbchangelog-3.4.xsd">
+
+    <!-- create db functions and stored procedures that will help with the db migration -->
+    <changeSet logicalFilePath="dummy" author="davidan" id="create_shard_functions_and_procedures">
+        <sql stripComments="true" endDelimiter="/">
+            -- function to produce a shard identifier from a workspace uuid string
+            -- important: the values produced by this function must match exactly the values produced by Rawls scala code
+            -- and the values inserted into 'shards' above
+            DROP FUNCTION IF EXISTS shardIdentifier /
+            CREATE FUNCTION shardIdentifier(workspaceId CHAR(32))
+            RETURNS CHAR(4) DETERMINISTIC
+            BEGIN
+                -- this function produces 32 shards
+                DECLARE stringToInspect CHAR(2);
+                DECLARE secondCharToInspect CHAR(1);
+                DECLARE shardIdPart1 CHAR(1);
+                DECLARE shardIdPart2 CHAR(2);
+
+                SET stringToInspect = SUBSTRING(workspaceId,1,2);
+                SET secondCharToInspect = SUBSTRING(stringToInspect,2,1);
+
+                -- first part of the shard id is equal to the first char of the workspace id, 16 possible values
+                SET shardIdPart1 = SUBSTRING(stringToInspect,1,1);
+                -- second part of the shard id produces 2 values: 0-7 and 8-f
+                IF secondCharToInspect >= '0' &amp;&amp; secondCharToInspect &lt;= '7' THEN
```

Comment:
I see `>=` and `&lt;=` on this line.. if liquibase lets us use > and < I'd say use them rather the xml escaped versions.